apps/src/templates/progress/ProgressLessonTeacherInfo.jsx

```diff
@@ -36,12 +36,12 @@ const ProgressLessonTeacherInfo = React.createClass({
     hiddenStageState: React.PropTypes.object.isRequired,
     scriptName: React.PropTypes.string.isRequired,
     hasNoSections: React.PropTypes.bool.isRequired,
-    toggleHidden: React.PropTypes.func.isRequired
+    toggleHiddenStage: React.PropTypes.func.isRequired
   },
 
   onClickHiddenToggle(value) {
-    const { scriptName, sectionId, lesson, toggleHidden } = this.props;
-    toggleHidden(scriptName, sectionId, lesson.id, value === 'hidden');
+    const { scriptName, sectionId, lesson, toggleHiddenStage } = this.props;
+    toggleHiddenStage(scriptName, sectionId, lesson.id, value === 'hidden');
```

Comment:
I love how this turned out, and also wish we could eliminate this plumbing. :grin:
![image](https://user-images.githubusercontent.com/1615761/29935562-0786cb46-8e34-11e7-9f07-7934fcab6101.png)